autoscale-groups-core/src/main/java/fish/payara/extensions/autoscale/groups/core/admin/ScaleCommandHelper.java

```diff
@@ -0,0 +1,327 @@
+            adminTimeout = 57 * 1000;
+        }
+
+        if (LOGGER.isLoggable(FINE)) {
+            LOGGER.fine(String.format("Initial cluster command timeout: %d ms", adminTimeout));
+        }
+
+        // Now go get results from the response queue.
+        for (int n = 0; n < nInstances; n++) {
+            long timeLeft = adminTimeout - (System.currentTimeMillis() - startTime);
+            if (timeLeft < 0) {
+                timeLeft = 0;
+            }
+            CommandRunnable cmdRunnable = null;
+            try {
+                cmdRunnable = responseQueue.poll(timeLeft, MILLISECONDS);
+            } catch (InterruptedException e) {
+                // This thread has been interrupted. Abort
+                threadPool.shutdownNow();
+                String msg = Strings.get("cluster.command.interrupted", targetNames, n, nInstances, commandName);
+                LOGGER.warning(msg);
+                failureOccurred = true;
+                // Re-establish interrupted state on thread
+                Thread.currentThread().interrupt();
+                break;
+            }
+
+            if (cmdRunnable == null) {
+                // We've timed out.
+                break;
+            }
+            String cname = cmdRunnable.getName();
+            waitingForServerNames.remove(cname);
+            ActionReport instanceReport = cmdRunnable.getActionReport();
+            if (LOGGER.isLoggable(FINE)) {
+                LOGGER.fine(String.format("Instance %d of %d (%s) has responded with %s", n + 1, nInstances, cname,
+                        instanceReport.getActionExitCode()));
+            }
+
+            if (instanceReport.getActionExitCode() != SUCCESS) {
+                // Bummer, the command had an error. Log and save output
+                failureOccurred = true;
+                failedServerNames.append(cname).append(" ");
+                reportResult.failedServerNames.add(cname);
+                String msg = cname + ": " + instanceReport.getMessage();
+                LOGGER.severe(msg);
+                msg = Strings.get("cluster.command.instancesFailed", commandName, cname);
+                progressStatus.progress(1, msg);
+            } else {
+                // Command worked. Note that too.
+                succeededServerNames.append(cname).append(" ");
+                reportResult.succeededServerNames.add(cname);
+                progressStatus.progress(1, cname);
+            }
+        }
+
+        actionReport.setActionExitCode(SUCCESS);
+
+        if (failureOccurred) {
+            actionReport.setResultType(List.class, reportResult.failedServerNames);
+        } else {
+            actionReport.setResultType(List.class, reportResult.succeededServerNames);
+        }
+
+        // Display summary of started servers
+        if (succeededServerNames.length() > 0) {
+            output.append("\n").append(Strings.get("cluster.command.instancesSucceeded", commandName,
```

Comment:
This may reference the wrong message string, see my previous comment

Reply:
`The command {0} executed successfully for: {1}`